src/analyses/variable-sensitivity/constant_pointer_abstract_object.cpp

```diff
@@ -72,6 +72,48 @@ abstract_object_pointert constant_pointer_abstract_objectt::merge(
   return abstract_pointer_objectt::merge(other, widen_mode);
 }
 
+bool constant_pointer_abstract_objectt::same_target(
+  abstract_object_pointert other) const
+{
+  auto cast_other =
+    std::dynamic_pointer_cast<const constant_pointer_abstract_objectt>(other);
+
+  if(value_stack.is_top_value() || cast_other->value_stack.is_top_value())
+    return false;
+
+  if(value_stack.depth() != cast_other->value_stack.depth())
+    return false;
+
+  for(size_t d = 0; d != value_stack.depth() - 1; ++d)
```

Comment:
Looks reasonable, the -1 is needed because you don't want the most detailed part of the offset.